### PR TITLE
Fix test case backlog.population

### PR DESCRIPTION
**Pull request by clemahieu**

### CODE REVIEW

File: nano/core_test/backlog.cpp [hunk 1]
```diff
@@ -72,10 +72,10 @@ block_list_t setup_independent_blocks (nano::test::system & system, nano::node &
  */
 TEST (backlog, population)
 {
```
Comment (dsiganos) [inline nano/core_test/backlog.cpp:74]: Moving the mutex upwards does not fix the problem. The mutex is still destroyed before the node. The node is destroyed much after the test case comes to an end.

**Comment by dsiganos** [comment]
Duplicate of #4059

File: nano/core_test/backlog.cpp (hunk 1 above)
Comment (pwojcikdev) [reply nano/core_test/backlog.cpp:74]: I don't think this is true. `system` destructor does indeed stop the nodes, which means no node threads should be running after that

**Review by pwojcikdev — DISMISSED** [review]

File: nano/node/backlog_population.cpp [hunk 2]
```diff
@@ -13,6 +13,7 @@ nano::backlog_population::backlog_population (const config & config_a, nano::sto
 
 nano::backlog_population::~backlog_population ()
 {
+	stop ();
```
Comment (pwojcikdev) [inline nano/node/backlog_population.cpp]: This shouldn't be there, while it's not an error, we want to ensure that when node destruction happens all threads have already been stopped.

**Review by pwojcikdev — APPROVED** [review]

**Review by dsiganos — APPROVED** [review]

**Pull request merged**